testing/src/main/groovy/io/opentelemetry/auto/test/asserts/TraceAssert.groovy

```diff
@@ -25,10 +25,7 @@ class TraceAssert {
                           @DelegatesTo(value = TraceAssert, strategy = Closure.DELEGATE_FIRST) Closure spec) {
     def spans = getTrace(writer, traceId)
     Stopwatch stopwatch = Stopwatch.createStarted()
-    while (stopwatch.elapsed(TimeUnit.SECONDS) < 10) {
-      if (spans.size() == expectedSize) {
-        break
-      }
+    while (spans.size() < expectedSize && stopwatch.elapsed(TimeUnit.SECONDS) < 10) {
```

Comment:
is there a way with spock to put a timeout on the test itself, rather than having to put it in the middle here?

Reply:
yes, but then we would have to apply timeout appropriately for each tests (and tests vary a lot in time since they are more integration test than unit test)